lib/spack/spack/build_environment.py

```diff
@@ -813,158 +801,242 @@ def setup_package(pkg, dirty, context="build"):
     return env_base
 
 
-def _make_runnable(pkg, env):
-    # Helper method which prepends a Package's bin/ prefix to the PATH
-    # environment variable
-    prefix = pkg.prefix
-
-    for dirname in ["bin", "bin64"]:
-        bin_dir = os.path.join(prefix, dirname)
-        if os.path.isdir(bin_dir):
-            env.prepend_path("PATH", bin_dir)
+class EnvironmentVisitor:
+    def __init__(self, *roots: spack.spec.Spec, context: Context):
+        # For the roots (well, marked specs) we follow different edges
+        # than for their deps, depending on the context.
+        self.root_hashes = set(s.dag_hash() for s in roots)
+
+        if context == Context.BUILD:
+            # Drop direct run deps in build context
```

Comment:
Yeah, it is a change of behavior.

Do we have good reasons to put them in the PATH?

A package like `tar` has `run` deps on `xz` / `bzip2` / ... to hard-code paths to `bin/*`. I guess another way of looking at that is that it's a `link/run` type dep, since the executable path is fixed instead of found in PATH.

Reply:
I think this is a good change.  If CI passes I'm good with it, and I think we should be making packages be more explicit about their dependencies -- even ones from configure.